katello/hooks/boot/13-hiera.rb

```diff
@@ -3,9 +3,11 @@
 TUNING_SIZES = ['default'] + ['medium', 'large', 'extra-large', 'extra-extra-large']
 TUNING_FACT = 'tuning'.freeze
 
-app_option(
-  '--tuning',
-  'INSTALLATION_SIZE',
-  "Tune for an installation size. Choices: #{TUNING_SIZES.join(', ')}",
-  :default => get_custom_fact(TUNING_FACT)
-)
+if module_enabled?('katello')
```

Comment:
I doubt `module_enabled?` actually works at the boot phase. https://github.com/theforeman/kafo#hooks suggests it shouldn't.

An alternative approach would be to see if `get_custom_fact` has a value. Then in migrations we can make sure a value is set in the katello scenario. Technically migrations aren't executed before this, but in the RPM there's a script that runs the migrations so for the end user it'd work.

Reply:
If I read the code correctly, the boot hook runs after the scenario and config have been both loaded and migrated which means kafo has in memory the migrated config and answers but prior to any parameters. The only parameters that could affect this would be `--no-enable` which we don't really support anyway. I tested this and encouraged @sthirugn to go this route. I'm running another test to double check my initial results and will report back.

https://github.com/theforeman/kafo/blob/master/lib/kafo/kafo_configure.rb#L89

Reply:
```
[root@installer vagrant]# foreman-installer --tuning
Insufficient memory for tuning size
Tuning profile 'default' requires at least 8 GB of memory and 1 CPU cores
[root@installer vagrant]# vi /usr/share/foreman-installer/katello/hooks/boot/13-hiera.rb 
[root@installer vagrant]# cat /usr/share/foreman-installer/katello/hooks/boot/13-hiera.rb 
# TODO: automatically get the tuning sizes - standard doesn't exist and only
# loads base. The rest maps to config/foreman.hiera/tuning/sizes/$size.yaml
TUNING_SIZES = ['default'] + ['medium', 'large', 'extra-large', 'extra-extra-large']
TUNING_FACT = 'tuning'.freeze

if module_enabled?('blah')
app_option(
  '--tuning',
  'INSTALLATION_SIZE',
  "Tune for an installation size. Choices: #{TUNING_SIZES.join(', ')}",
  :default => get_custom_fact(TUNING_FACT)
)
end
[root@installer vagrant]# foreman-installer --tuning
ERROR: Unrecognised option '--tuning'

See: 'foreman-installer --help'
[root@installer vagrant]# vi /usr/share/foreman-installer/katello/hooks/boot/13-hiera.rb 
[root@installer vagrant]# cat /usr/share/foreman-installer/katello/hooks/boot/13-hiera.rb 
# TODO: automatically get the tuning sizes - standard doesn't exist and only
# loads base. The rest maps to config/foreman.hiera/tuning/sizes/$size.yaml
TUNING_SIZES = ['default'] + ['medium', 'large', 'extra-large', 'extra-extra-large']
TUNING_FACT = 'tuning'.freeze

if module_enabled?('katello')
app_option(
  '--tuning',
  'INSTALLATION_SIZE',
  "Tune for an installation size. Choices: #{TUNING_SIZES.join(', ')}",
  :default => get_custom_fact(TUNING_FACT)
)
end
[root@installer vagrant]# foreman-installer --tuning
Insufficient memory for tuning size
Tuning profile 'default' requires at least 8 GB of memory and 1 CPU cores
[root@installer vagrant]# 
```

Reply:
> If I read the code correctly, the boot hook runs after the scenario and config have been both loaded and migrated which means kafo has in memory the migrated config and answers but prior to any parameters.

That would be good. I only read the documentation and that gave me the impression it might not work.

> The only parameters that could affect this would be `--no-enable` which we don't really support anyway.

The alternative is `unless kafo.module('katello').nil?` which only checks if the module is present, regardless of the exact answer.

It should be rather easy to verify because you can run the installer locally and `--help` should work. You can also add a line line:
https://github.com/theforeman/foreman-installer/blob/20a1c2d28b798cf8cb67e8c861149cc56047fc58/.travis.yml#L29
Though we already have the output in Travis because we have the generic `--help`. I only see `--tuning` for Katello so I think that confirms this works. We have more CI than we think ;)

Can't we check for the `foreman` module instead of `katello`? That's probably more accurate since our tuning profiles are more to Foreman/Apache/PostgreSQL.

Reply:
Before submitting the PR, I tested the code locally in both 6.7 satellite and capsule servers with the changes from this PR:

```console
# hostname
sat-6-7-qa-rhel7.apex.example.com

# satellite-installer --scenario satellite --help | grep 'tuning'
    --tuning INSTALLATION_SIZE    Tune for an installation size. Choices: default, medium, large, extra-large, extra-extra-large (default: "medium")
```
```console
# hostname
capsule-6-7-qa-rhel7.apex.example.com

# satellite-installer --scenario capsule --help | grep 'tuning'
<No Output>
```